sycl/test/regression/private_array_init_test.cpp

```diff
@@ -0,0 +1,50 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+//==- private_array_init_test.cpp - Regression test for private array init -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+namespace s = cl::sycl;
+
+class A {
+public:
+  A() : _arr{0, 0, 0} {}
+
+  size_t size() {
+    return sizeof(_arr) / sizeof(_arr[0]);
+  }
+
+private:
+  size_t _arr[3];
+};
+
+int main() {
+  size_t data = 1;
+
+  s::queue q;
+  {
+    s::buffer<size_t, 1> buf(&data, s::range<1>(1));
+    q.submit([&](s::handler &cgh) {
+      auto acc = buf.get_access<s::access::mode::read_write>(cgh);
+      cgh.single_task<class test>([=]() {
+        // Test that device code reverse translation from SPIR-V is not failed
```

Comment:
If it's about reverse translation, why the test isn't placed in the translator's repository? There are external participants here, like Xilinx, who aren't relaying on SPIR-V and therefore it would be hard for them to maintain this test,

Reply:
Comment has been rewritten in order to not confuse people by details of compilation specifics for one product.